deebotozmo/cli.py

```diff
@@ -0,0 +1,439 @@
+    vacbot.bot.events.stats.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+
+@cli.command(help="Get robot components life span")
+@coro
+async def components() -> None:
+    """Click subcommand that returns the robot's life span."""
+    vacbot = DoLogin()
+    await vacbot.run()
+
+    lock = asyncio.Lock()
+    await lock.acquire()
+
+    async def on_lifespan_event(event: dict) -> None:
+        for key, value in event.items():
+            print(f"{key}: {value}%")
+        lock.release()
+
+    listener = vacbot.bot.events.lifespan.subscribe(on_lifespan_event)
+    await lock.acquire()
+    vacbot.bot.events.lifespan.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+
+@cli.command(name="getrooms", help="Get saved rooms")
+@coro
+async def get_rooms() -> None:
+    """Click subcommand that returns saved room."""
+    vacbot = DoLogin()
+    await vacbot.run()
+
+    lock = asyncio.Lock()
+    await lock.acquire()
+
+    async def on_rooms(event: RoomsEvent) -> None:
+        for room in event.rooms:
+            print(str(room.id) + " " + str(room.subtype))
+        lock.release()
+
+    listener = vacbot.bot.map.events.rooms.subscribe(on_rooms)
+    await lock.acquire()
+    vacbot.bot.map.events.rooms.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+
+@cli.command(
+    name="exportlivemap",
+    help='Get robot map and save it [filepath ex: "/folder/livemap.png"',
+)
+@click.argument("filepath", type=click.STRING, required=True)
+@coro
+async def export_live_map(filepath: str) -> None:
+    """Click subcommand that returns the live map."""
+    vacbot = DoLogin()
+    await vacbot.run()
+
+    lock = asyncio.Lock()
+    await lock.acquire()
+
+    async def on_map(_: MapEvent) -> None:
+        with open(filepath, "wb") as file:
+            file.write(base64.decodebytes(vacbot.bot.map.get_base64_map()))
+        lock.release()
+
+    listener = vacbot.bot.events.map.subscribe(on_map)
+    await lock.acquire()
+    vacbot.bot.events.map.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+
+class DoLogin:
+    """DoLogin communicates with the bot to provide self.bot."""
+
+    def __init__(self) -> None:
+        """Init DoLogin."""
+        # self.session = None
+        # self.auth = None
+        # self.devices_ = None
+        # self.bot = None
+        # self.mqtt = None
+        return
+
+    async def run(self) -> None:
+        # pylint: disable=attribute-defined-outside-init
+        """Communicate with Deebot."""
+        config = read_config()
+
+        self.session = aiohttp.ClientSession()
+
+        self.api = EcovacsAPI(
+            self.session,
+            config["device_id"],
+            config["email"],
+            config["password_hash"],
+            continent=config["continent"],
+            country=config["country"],
+            verify_ssl=bool(config["verify_ssl"]),
+        )
+
+        if not config_file_exists():
+            click.echo(
+                "Not logged in. "
```

Comment:
```suggestion
                f"Not logged in. Do '{os.path.basename(sys.argv[0])} createconfig' first."
```
Do not split it on two lines

Reply:
Otherwise `pylint` complains about the line being too long

Reply:
Please run first `pre-commit run --hook-stage manual --all-files` and afterwards `pylint`. So blacks format the file correctly

Reply:
never mind it no longer complains, fixed